arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -38,3 +38,10 @@ config NXP_MPU
 	default n
 	help
 	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
+	bool "Add MPU access to write to flash"
+	depends on NXP_MPU
+	default n
```

Comment:
To be a little more precise:
If MPU is enabled (or future task isolation, etc), the default for all added options should be to maintain the current level of security.  The default should never relax security.